packages/frontend/src/components/create-campaign/preview/index.tsx

```diff
@@ -144,11 +153,75 @@ export function CampaignPreview({
         if (payload.kpiSpecification && rewardsApproved) uploadSpecification();
     }, [payload.kpiSpecification, rewardsApproved]);
 
+    const handleSafeTransaction = useCallback((tx: BaseTransaction) => {
+        setSafeTxs((txs) => {
+            txs.push(tx);
+            return txs;
+        });
+    }, []);
+
     function handleOnRewardsApproved() {
         setRewardsApproved(true);
     }
 
+    // TODO: separate logic and function for Safe context and standard context
     const handleOnDeploy = useCallback(() => {
+        if (safeContext) {
+            if (
+                !rewardsApproved ||
+                !chainData?.metromContract.address ||
+                !payload.pool ||
+                !payload.startDate ||
+                !payload.endDate ||
+                !payload.rewards ||
+                payload.rewards.length === 0
+            ) {
+                console.warn(
+                    "Missing parameters to deploy campaign through Safe: aborting",
+                );
+                return;
+            }
+
+            safeTxs.push({
+                to: chainData.metromContract.address,
+                data: encodeFunctionData({
+                    abi: metromAbi,
+                    functionName: "createCampaigns",
+                    args: [
+                        [
+                            {
+                                pool: payload.pool.address,
+                                from: payload.startDate.unix(),
+                                to: payload.endDate.unix(),
+                                specification: specificationHash,
+                                rewards: payload.rewards.map((reward) => ({
+                                    token: reward.token.address,
+                                    amount: reward.amount.raw,
+                                })),
+                            },
+                        ],
+                    ],
+                }),
+                value: "0",
+            });
+
+            const create = async () => {
+                setDeploying(true);
+                try {
+                    await safeSdk.txs.send({ txs: safeTxs });
+                    trackFathomEvent("CLICK_DEPLOY_CAMPAIGN");
+                } catch (error) {
+                    console.warn("could not create kpi token", error);
```

Comment:
Is the warn message appropriate here?

Reply:
Addressed in 849c227d6bcc5b773cbffca69dce9732881cbb51